aws-xray-recorder-sdk-spring/src/main/java/com/amazonaws/xray/spring/aop/AbstractXRayInterceptor.java

```diff
@@ -0,0 +1,95 @@
+package com.amazonaws.xray.spring.aop;
+
+import com.amazonaws.xray.AWSXRay;
+import com.amazonaws.xray.entities.Subsegment;
+import com.amazonaws.xray.exceptions.SegmentNotFoundException;
+import com.amazonaws.xray.strategy.ContextMissingStrategy;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Pointcut;
+
+public abstract class AbstractXRayInterceptor {
+
+    private static final Log logger = LogFactory.getLog(AbstractXRayInterceptor.class);
+
+    private static ContextMissingStrategy getContextMissingStrategy() {
+        return AWSXRay.getGlobalRecorder().getContextMissingStrategy();
+    }
+
+    /**
+     * @param pjp the proceeding join point
+     * @return the result of the method being wrapped
+     * @throws Throwable
+     */
+    @Around("xrayTracedClasses() || xrayEnabledClasses()")
+    public Object traceAroundMethods(ProceedingJoinPoint pjp) throws Throwable {
+        return this.processXRayTrace(pjp);
+    }
+
+    protected Object processXRayTrace(ProceedingJoinPoint pjp) throws Throwable {
+        boolean endSegment = false;
+        try {
+            if (AWSXRay.getCurrentSegment() != null) {
+                logger.trace("Current segment exists");
+            }
+        } catch (SegmentNotFoundException snfe) {
```

Comment:
Sorry if my previous comment regarding context missing strategies was unclear. The logic inside the `getCurrentSegment()` function should reference the context missing strategy as needed - and will only throw the `SegmentNotFoundException` if the strategy calls for it.

```java
    public Segment getCurrentSegment() {
        Optional<Segment> segment = getCurrentSegmentOptional();
        if (segment.isPresent()) {
            return segment.get();
        }
        contextMissingStrategy.contextMissing("No segment in progress.", SegmentNotFoundException.class);
        return null;
    }
```

So, there should be no need to catch exceptions here. And the `boolean endSegment` shouldn't be required either. We can call `getSegment()` and compare the value against `null` to determine whether or not a subsegment should be created here.